### PR TITLE
feat: Implementing Github Sign In OAuth 

### PR DESCRIPTION
**Pull request by swap-musale**

### CODE REVIEW

File: app/src/main/java/com/example/githubcloneapp/util/Constant.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.example.githubcloneapp.util
+
+object Constant {
+
+    const val REST_BASE_URL = "https://github.com/"
+    const val GRAPH_QL_BASE_URL = "https://api.github.com/graphql"
+    const val APP_CLIENT_ID = "8d495907d9bdbe7f5388"
```
Comment (pushpalroymm) [inline app/src/main/java/com/example/githubcloneapp/util/Constant.kt:7]: @swap-musale Can we pass these values from Github Secrets? In that way, we don't have to hardcode these.

File: domain/src/main/java/com/example/domain/useCase/GetUserInfoUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.example.domain.useCase
+
+import com.example.domain.repository.GithubRepository
+
+class GetUserInfoUseCase(private val githubRepository: GithubRepository) {
```
Comment (pushpalroymm) [inline domain/src/main/java/com/example/domain/useCase/GetUserInfoUseCase.kt:5]: @swap-musale Can we create a BaseUseCase for all use cases?

File: domain/src/main/java/com/example/domain/useCase/GetUserAccessToken.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.example.domain.useCase
+
+import com.example.domain.model.AccessTokenRequest
+import com.example.domain.repository.GithubRepository
+
+class GetUserAccessToken(private val githubRepository: GithubRepository) {
```
Comment (pushpalroymm) [inline domain/src/main/java/com/example/domain/useCase/GetUserAccessToken.kt:6]: @swap-musale Append with UseCase.

File: domain/src/main/java/com/example/domain/model/UserInfo.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.example.domain.model
+
+data class UserInfo(
+    val name: String,
```
Comment (pushpalroymm) [inline domain/src/main/java/com/example/domain/model/UserInfo.kt:4]: @swap-musale Are all these values coming from server API calls? In that case, shouldn't we make these nullable?

**Review by pushpalroymm — CHANGES_REQUESTED** [review]

**Comment by swap-musale** [comment]
closing this  PR Code as it is outdated and same functionality code available in [this PR](https://github.com/swap-musale/GithubCloneApp/pull/2).

**Pull request closed without merging**